src/Payum/Core/CoreGatewayFactory.php

```diff
@@ -59,10 +63,26 @@ public function createConfig(array $config = array())
         $config['twig.env'] = null;
 
         $config->defaults(array(
-            'payum.template.layout' => '@PayumCore/layout.html.twig',
+            'httplug.client'=>function(ArrayObject $config) {
+                return HttpClientDiscovery::find();
+            },
+            'httplug.message_factory'=>function(ArrayObject $config) {
+                return MessageFactoryDiscovery::find();
+            },
+            'payum.http_client'=>function(ArrayObject $config) {
+                $func = $config['httplug.client'];
+                if (is_callable($func)) {
+                    $func = $func($config);
+                }
```

Comment:
@Nyholm you dont have to do it here since `httplug.client` will be resolved before the payum.http_client [here](https://github.com/Payum/Payum/pull/505/files#diff-d6a9354452023166c9054c19a425a2c8R134).